Update Selenium packages for Null Pointer vulnerability

egottliebq2 opened the pull request:
## Description

A [null pointer vulnerability](https://github.com/seleniumhq/selenium/commit/023a0d52f106321838ab1c0997e76693f4dcbdf6) was found in Selenium in versions prior to 4.14.0 and fixed in version 4.14.1. Updated the corresponding Selenium packages to the latest stable version and corrected the obsoletion of `ScreenshotImageFormat` that occurred in version 4.14.1 to keep current compatibility and limit changes.


### Changes

- Updated `Selenium.Support` and `Selenium.Webdriver` to Version 4.28.0
- `ScreenshotImageFormat` is obsolete within `OpenQA.Selenium` as of v4.14.0. Adjusted use-cases for screenshot image formatting to append the string format to the end of the file path. Adjusted corresponding unit tests to match new method conditions.

## Testing

### Unit Tests
![image](https://github.com/user-attachments/assets/76e968f5-b671-48f6-abc1-5f98622f7e9a)


## Checklist

- [x] I agree to follow Boa Constrictor's [Code of Conduct](https://q2ebanking.github.io/boa-constrictor/contributing/code-of-conduct/).
- [x] I read Boa Constrictor's [Contributing Code](https://q2ebanking.github.io/boa-constrictor/contributing/contributing-code/) guide.
- [x] I successfully built the .NET solution with no errors or new warnings.
- [x] I successfully ran both the unit tests and the example tests.
- [x] I updated the [appropriate changelogs](CHANGELOG.md) with concise descriptions of these changes.
- [x] I added documentation for these changes (if appropriate).

AutomationPanda reviewed (changes requested):
It looks like the automated tests are failing. Please fix these. Otherwise, the code looks okay.

egottliebq2 commented:
## Additional Changes Description

All unit tests were failing due to a [deprecated version of the artifact actions](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/) as well as an out of date version of .NET. The github action workflow for unit tests had to be updated to address this deprecation, as well as each of the `.UnitTests` regions.


### Changes

- Changed target framework for the unit tests and example projects to .NET 8
- Changed github artifact actions from `v3` to `v4`

## Testing

### Unit Tests
![image](https://github.com/user-attachments/assets/41b92052-fefb-4c20-ab19-e782830a6511)


## Checklist

- [x] I agree to follow Boa Constrictor's [Code of Conduct](https://q2ebanking.github.io/boa-constrictor/contributing/code-of-conduct/).
- [x] I read Boa Constrictor's [Contributing Code](https://q2ebanking.github.io/boa-constrictor/contributing/contributing-code/) guide.
- [x] I successfully built the .NET solution with no errors or new warnings.
- [x] I successfully ran both the unit tests and the example tests.
- [x] I updated the [appropriate changelogs](CHANGELOG.md) with concise descriptions of these changes.
- [x] I added documentation for these changes (if appropriate).

pl-shernandez reviewed (approved):
@egottliebq2 Ran unit tests one more time looks good 👍
